daphne_worker/src/durable/reports_processed.rs

```diff
@@ -185,15 +198,16 @@ impl ReportsProcessed {
             // Output: `Vec<ReportId>`
             (DURABLE_REPORTS_PROCESSED_MARK_AGGREGATED, Method::Post) => {
                 let report_ids: Vec<ReportId> = req_parse(&mut req).await?;
-                let replayed_reports = try_join_all(
-                    report_ids
-                        .into_iter()
-                        .map(|report_id| self.to_checked(report_id)),
-                )
-                .await?
-                .into_iter()
-                .flatten()
-                .collect::<Vec<ReportId>>();
+                let mut seen = self.load_seen_reports().await?;
```

Comment:
High level question: Isn't this going to get expensive if the number of reports processed gets large? Do we have a sense of how many is too many? Once we reach that point, what should we do at the protocol level?

Reply:
I tested for more than 8k reports and it behaved just fine. But perhaps we need to measure what a real workload looks like

Reply:
Did you test this in `wrangler dev` or on Workers?

To avoid creating a DoS vector here, we're going to want to enforce a cap on the number of reports we're willing to store in a single instance. If we overflow this, then we'll need to reject reports for which we aren't tracking replay (with [PrepareError "report_dropped"](https://www.ietf.org/archive/id/draft-ietf-ppm-dap-05.html#section-4.5.1.2-5)). Note that the "mark aggregated" call back will have to handle the rejection logic.